integration-tests/goss/goss-shared.yaml

```
@@ -211,3 +211,9 @@ matching:
         - have-key-with-value:
             foo: bar
         - have-key: baz
+  semver:
+    content:
+      - 1.0.1
+      - 1.9.9
+    matches:
+      semver-constraint: ">1.0.0 <2.0.0 !=1.5.0"
```

```suggestion
      semver-constraint: ">1.0.0 <2.0.0 !=1.5.0"
  semver2:
    content:
      - 1.0.1
      - 1.5.0
      - 1.9.9
    matches:
      not:
        semver-constraint: ">1.0.0 <2.0.0 !=1.5.0"
```

This adds a test proving the negation works. It might not be needed given all the unit tests.. guess I'm just being a little extra thorough for now.

The fact that you added unit tests for this is greatly appreciated by the way. Hoping with time goss moves to depending more on unit tests for stuff like this.

Side note, @OmarDarwish thanks for the `matching` resource, fit great for this test! Lightbulb just turned on for me on how useful it is for testing/debugging matchers.

---

Added manually

---

By all means, extra tests are always a good thing!

---

>Side note, @OmarDarwish thanks for the matching resource, fit great for this test! Lightbulb just turned on for me on how useful it is for testing/debugging matchers.

Glad it's providing use!